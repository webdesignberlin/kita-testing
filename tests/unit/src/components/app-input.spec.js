import { mount } from '@vue/test-utils';
import AppInput from '../../../../src/components/app-input.vue'

test('mount component', async() => {
  expect(AppInput).toBeTruthy()

  const wrapper = mount(AppInput, {
    props: {
      type: 'password',
      modelValue: 'testi',
    },
  });

  const input = await wrapper.get('input');
  expect(input.element.type).toEqual('password');
  expect(input.element.value).toEqual('testi');
  await input.setValue('testit');
  expect(wrapper.emitted()).toBeTruthy();
  expect(wrapper.emitted()).toHaveProperty('update:modelValue');
  expect(input.element.value).toEqual('testit');
  expect(wrapper.html()).toMatchSnapshot()
});
