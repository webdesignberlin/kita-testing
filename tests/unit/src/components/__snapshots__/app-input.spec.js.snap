// Vitest Snapshot v1

exports[`mount component 1`] = `"<label class=\\"field\\" data-v-46a1db84=\\"\\"><span class=\\"field__text\\" data-v-46a1db84=\\"\\"></span><input type=\\"password\\" class=\\"field__input\\" data-v-46a1db84=\\"\\" /></label>"`;
